Return a list of files ready to be read

milljm opened the pull request:
Instead of returning False or True, return a list of available files. The
issue with no expected out, was due to having just one output not available.

Take the following example:

  ./run_tests -p 4

Process 2 has the output we need to regex (RunException), but process 3, was
killed by MPI, and never created a file. This caused checkOutputReady to
return false. Yet the other processes successfully created output, and thus,
would have been found by RunException.

This issue was most apparent on Virtual Machines with slow filesystems:

  https://www.moosebuild.org/job/147018/

However, we rarely also saw this on our build machines (invalidate, and then
the test magically passes later). Finally... a fix!

The change to the Tester (outfie, errfile) .flush was something recommended
by Brian, that might solve this issue. Leaving it in there, because we should
be flushing the buffer anyways.

Closes #10570

Commit:
Return a list of files ready to be read

Instead of returning False or True, return a list of available files. The
issue with no expected out, was due to having just one output not available.

Take the following example:

  ./run_tests -p 4

Process 2 has the output we need to regex (RunException), but process 3, was
killed by MPI, and never created a file. This caused checkOutputReady to
return false. Yet the other processes successfully created output, and thus,
would have been found by RunException.

This issue was most apparent on Virtual Machines with slow filesystems:

  https://www.moosebuild.org/job/147018/

However, we rarely also saw this on our build machines (invalidate, and then
the test magically passes later). Finally... a fix!

The change to the Tester (outfie, errfile) .flush was something recommended
by Brian, that might solve this issue. Leaving it in there, because we should
be flushing the buffer anyways.

Closes #10570

moosebuild commented:
Job [Documentation](https://www.moosebuild.org/job/147613/) on 78f0b7d wanted to post the following:

View the site [here](http://mooseframework.org/docs/PRs/10571/site)

This comment will be updated on new commits.

Commit:
Print file name instead of processor

Because we are possibly returning an incomplete list,
print the filename, instead of the enumerated index
integer as the processor responsible for the file being
read.

Closes #10570

milljm commented:
@brianmoose I 'Canceled' Parallel Sweep target before committing a new PR. But it looks like it still wants to run the Parallel Sweep target. It looks as though Distributed Mesh Sweep is not going to run again. So 'Cancel' partially worked.
